### PR TITLE
Update changelog for AppConfig release

### PR DESCRIPTION
**Pull request by antkmsft**
Release work item: https://dev.azure.com/azure-sdk/Release/_workitems/edit/11793/

### CODE REVIEW

File: sdk/appconfig/azappconfig/CHANGELOG.md [hunk 1]
```diff
@@ -1,16 +1,10 @@
 # Release History
 
-## 0.3.0 (Unreleased)
-
-### Features Added
+## 0.3.0 (2022-05-10)
```
Comment (jhendrixMSFT) [inline sdk/appconfig/azappconfig/CHANGELOG.md:3]: I'm hoping to release this today once `azcore` release is done.

Comment (antkmsft) [reply sdk/appconfig/azappconfig/CHANGELOG.md:3]: We sure can! In general, Tuesday is the release date, but if releasing today is good for our customers, we should do it! I'll make a suggestion edit to change the date to 5/9, and I'll keep monitoring for the appearance of the Core release. Then, please approve this PR, and you can even submit the PR proposal with the date, and I'll trigger the release pipelines.

Comment (antkmsft) [inline sdk/appconfig/azappconfig/CHANGELOG.md:3]: @jhendrixMSFT, if we release today, I'll submit this change to the PR before triggering the release pipelines. I'll keep monitoring for the `azcore` release to appear before I start. But I'll need your approval for this PR, should we release today or tomorrow.
```suggestion
## 0.3.0 (2022-05-09)
```

**Comment by jhendrixMSFT** [comment]
Looks like the release of `azcore` will happen on 5/10 now.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**